src/parser/statement.js

```diff
@@ -666,17 +670,21 @@ pp.parseClassBody = function (node) {
 
     this.parsePropertyName(method);
 
-    method.static = isMaybeStatic && !this.match(tt.parenL);
+    let isMaybeGetSet = !method.computed && (method.key.name === "get" || method.key.name === "set");
+
+    method.static = isMaybeStatic && !this.match(tt.parenL) && !(this.match(tt.braceR) && this.hasPlugin("classProperties"));
     if (method.static) {
       if (isGenerator) this.unexpected();
       isGenerator = this.eat(tt.star);
       this.parsePropertyName(method);
     }
 
     if (!isGenerator) {
-      if (this.isClassProperty()) {
-        classBody.body.push(this.parseClassProperty(method));
-        continue;
+      if (this.isInitializedClassProperty() || (this.isUninitializedClassProperty() && !(isMaybeGetSet && this.match(tt.name)))) {
+        if (this.hasPlugin("classProperties") || this.hasPlugin("flow")) {
```

Comment:
I am not involved in the project, so take this with a grain of salt, but nested `if`s like this look strange to me - why not combine the two conditions?